### PR TITLE
Unit test for non-zero exit codes

### PR DESCRIPTION
**Pull request by jonocodes**
Making sure list commands return 1 when there is an error instead of 0. Addresses https://github.com/openfaas/faas-cli/issues/183

## Description

For example:

```
$ ./faas-cli list --gateway=http://bad-url

Get http://bad-url/system/functions: dial tcp: lookup bad-url on 127.0.0.1:53: server misbehaving
2017/10/31 13:23:27 cannot connect to OpenFaaS on URL: http://bad-url

$ echo $?
0
```

## Motivation and Context

- [ ] I have raised an issue to propose this change ([required](https://github.com/openfaas/faas/blob/master/CONTRIBUTING.md))

## How Has This Been Tested?

Manual and unit test.

Go 1.8.5

Changes are isolated to list function.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I've read the [CONTRIBUTION](https://github.com/openfaas/faas/blob/master/CONTRIBUTING.md) guide
- [x] I have signed-off my commits with `git commit -s`
- [x] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Comment by jonocodes** [comment]
Sorry. I'm new to go.
I need to figure out why this is failing on unit tests.

**Comment by alexellis** [comment]
Hey I saw the error log on Travis but wasn't sure what went wrong. What error do you get when you run this on your machine? There is a Makefile which can run the unit tests for you.

**Comment by jonocodes** [comment]
I spent an hour or two yesterday trying to figure this out, but I have only been able to reduce it to a conflict between these two tests.
`go test ./commands -run 'Test_remove|Test_list_errors' -v`

I thought something was going on with the mockserver, but Test_list_errors does not even use that anymore.

**Comment by alexellis** [comment]
Ok let's work together on this. Can you ping in #contributors?

**Comment by ericstoekl** [comment]
Here's your issue:

```
> github.com/openfaas/faas-cli/commands.runDelete() ./remove.go:41 (PC: 0x8d9de2)
    36:	  faas-cli remove img2ansi --gateway==http://remote-site.com:8080`,
    37:		Run: runDelete,
    38:	}
    39:	
    40:	func runDelete(cmd *cobra.Command, args []string) {
=>  41:		var services stack.Services
    42:		if len(yamlFile) > 0 {
    43:			parsedServices, err := stack.ParseYAMLFile(yamlFile, regex, filter)
    44:			if err != nil {
    45:				log.Fatalln(err.Error())
    46:				return
(dlv) p yamlFile
"non-existant-yaml"
```

The `yamlFile` is still set to `"non-existant-yaml"` when the `remove_Test` module runs. We need to reset this global.

**Comment by alexellis** [comment]
This is why I don't like global variables.. it create weird timing issues. Glad you could help Eric thank you.

File: commands/faas.go [hunk 1]
```diff
@@ -31,6 +31,13 @@ var (
 	language     string
 )
 
+// TODO: remove this workaround once these vars are no longer global
```
Comment (alexellis) [inline commands/faas.go]: @ericstoekl can we do this from within the tests instead? Would it work because I don't like this here, sorry.

Comment (ericstoekl) [reply commands/faas.go]: Yes, the usage of it is only in the test files anyway, so it should be in `test/utils.go`

Comment (jonocodes) [reply commands/faas.go]: My intent was to put in utils originally, but I could not access the variables from there. Dont know much about go, but seemed like a package scoping issue. Didnt want to get too invasive. Having to refactor or reflect.

**Comment by alexellis** [comment]
No need to put this in a central location for the PR. With the scope of the test file will be OK, then we can move forward and merge.

**Comment by johnmccabe** [comment]
@jonocodes looks like #207 has addressed this? 😕

**Comment by jonocodes** [comment]
Ok @alexellis I have updated this with master. Now it is only my test.

**Comment by alexellis** [comment]
Hi Jono, I'm trying to merge this for you but getting an error.

> This branch cannot be rebased due to conflicts
> Rebasing the commits of this branch on top of the base branch cannot be performed automatically due to conflicts encountered while reapplying the individual commits from the head branch.

If you pull down master from GitHub and reapply these changes in a text editor, I can merge in a new PR. I'll close this one but would like the change and to be able to attribute it to yourself.

**Comment by jonocodes** [comment]
sure @alexellis . moved it to https://github.com/openfaas/faas-cli/pull/229
though there is some travis issue with mongo there

**Comment by alexellis** [comment]
Thanks for working on this @jonocodes - hope to see you again in 2018 on Slack or a contributor meeting?

**Comment by jonocodes** [comment]
Sure, yup.

**Pull request closed without merging**